demos/common/processing/async_pipeline.h

```diff
@@ -0,0 +1,150 @@
+/*
+// Copyright (C) 2018-2020 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+*/
+
+#define MOVING_AVERAGE_SAMPLES 5
+
+#pragma once
+#include <string>
+#include <deque>
+#include <map>
+#include <condition_variable>
+#include "config_factory.h"
+#include "requests_pool.h"
+#include "results.h"
+#include "model_base.h"
+
+/// This is base class for asynchronous pipeline
+/// Derived classes should add functions for data submission and output processing
+class PipelineBase
+{
+public:
+    struct PerformanceInfo
+    {
+        int64_t framesCount = 0;
+        std::chrono::steady_clock::duration latencySum = std::chrono::steady_clock::duration::zero();
+        std::chrono::steady_clock::duration lastInferenceLatency = std::chrono::steady_clock::duration::zero();
+        std::chrono::steady_clock::time_point startTime;
+        double movingAverageLatencyMs;
+        uint32_t numRequestsInUse;
+        double movingAverageFPS;
+        double FPS=0;
```

Comment:
```suggestion
        double fps = 0;
```

Reply:
FPS is an acronym, we should keep letters capital

Reply:
I disagree. Acronyms in identifiers should be treated as regular words. Uppercase names should only be used for constants.

Reply:
there's no any definition for that case in coding style AFAIR, so let's keep em as is

Reply:
will be changed along with switching to PreformanceMetrics